sim/config/infrastructure/SpringbokRiscV32.cs

```diff
@@ -333,20 +333,22 @@ private void DefineRegisters()
                             Machine.LocalTimeSource.ExecuteInNearestSyncedState( __ => {
                                 if (InitMemSelect.Value)
                                 {
-                                    for(long writeAddress = InitStartAddress.Value & ~(InstructionPageSize - 1);
-                                writeAddress < ((InitEndAddress.Value + InstructionPageSize - 1) & ~(InstructionPageSize - 1));
-                                writeAddress += InstructionPageSize)
+                                    var instructionPageMask = ~((ulong)(InstructionPageSize - 1));
+                                    for(ulong writeAddress = InitStartAddress.Value & instructionPageMask;
+                                        writeAddress < ((InitEndAddress.Value + InstructionPageSize - 1) & instructionPageMask);
+                                        writeAddress += InstructionPageSize)
                                     {
-                                        IMem.WriteBytes(writeAddress, InstructionErasePattern, 0, InstructionPageSize);
+                                        IMem.WriteBytes((long)writeAddress, InstructionErasePattern, 0, InstructionPageSize);
```

Comment:
ulong?

Reply:
Actually this is correct - the `writeAddress` is of type ulong before the cast, but due to the limitations of array indexing we must use the signed value